tests/test_config.py

```diff
@@ -519,3 +543,180 @@ def test_run_with_ignored_path(self):
+            'ign-trail/file.yaml': bad_yaml,
+            'include/ign-dup/sub/dir/file.yaml': bad_yaml,
+            's/s/ign-trail/file.yaml': bad_yaml,
+            's/s/ign-trail/s/s/file.yaml': bad_yaml,
+            's/s/ign-trail/s/s/file2.lint-me-anyway.yaml': bad_yaml,
+            '.gitignore': ignored_files,
+
+            '.yamllint': 'ignore-from-file: .gitignore\n'
+                         'extends: default\n'
+        })
+
+        cls.backup_wd = os.getcwd()
+        os.chdir(cls.wd)
+
+    @classmethod
+    def tearDownClass(cls):
+        super(IgnoreFromFileConfigTestCase, cls).tearDownClass()
+
+        os.chdir(cls.backup_wd)
+
+        shutil.rmtree(cls.wd)
+
+    def test_run_with_ignored_from_file(self):
+        sys.stdout = StringIO()
+        with self.assertRaises(SystemExit):
+            cli.run(('-f', 'parsable', '.'))
+
+        out = sys.stdout.getvalue()
+        out = '\n'.join(sorted(out.splitlines()))
+
+        docstart = '[warning] missing document start "---" (document-start)'
+        keydup = '[error] duplication of key "key" in mapping (key-duplicates)'
+        trailing = '[error] trailing spaces (trailing-spaces)'
+        hyphen = '[error] too many spaces after hyphen (hyphens)'
+
+        self.assertEqual(out, '\n'.join((
+            './.yamllint:1:1: ' + docstart,
+            './bin/file.lint-me-anyway.yaml:3:3: ' + keydup,
+            './bin/file.lint-me-anyway.yaml:4:17: ' + trailing,
+            './bin/file.lint-me-anyway.yaml:5:5: ' + hyphen,
+            './file-at-root.yaml:3:3: ' + keydup,
+            './file-at-root.yaml:4:17: ' + trailing,
+            './file-at-root.yaml:5:5: ' + hyphen,
+            './ign-dup/file.yaml:3:3: ' + keydup,
+            './ign-dup/file.yaml:4:17: ' + trailing,
+            './ign-dup/file.yaml:5:5: ' + hyphen,
+            './ign-dup/sub/dir/file.yaml:3:3: ' + keydup,
+            './ign-dup/sub/dir/file.yaml:4:17: ' + trailing,
+            './ign-dup/sub/dir/file.yaml:5:5: ' + hyphen,
+            './ign-trail/file.yaml:3:3: ' + keydup,
+            './ign-trail/file.yaml:4:17: ' + trailing,
+            './ign-trail/file.yaml:5:5: ' + hyphen,
+            './include/ign-dup/sub/dir/file.yaml:3:3: ' + keydup,
+            './include/ign-dup/sub/dir/file.yaml:4:17: ' + trailing,
+            './include/ign-dup/sub/dir/file.yaml:5:5: ' + hyphen,
+            './s/s/ign-trail/file.yaml:3:3: ' + keydup,
+            './s/s/ign-trail/file.yaml:4:17: ' + trailing,
+            './s/s/ign-trail/file.yaml:5:5: ' + hyphen,
+            './s/s/ign-trail/s/s/file.yaml:3:3: ' + keydup,
+            './s/s/ign-trail/s/s/file.yaml:4:17: ' + trailing,
+            './s/s/ign-trail/s/s/file.yaml:5:5: ' + hyphen,
+            './s/s/ign-trail/s/s/file2.lint-me-anyway.yaml:3:3: ' + keydup,
+            './s/s/ign-trail/s/s/file2.lint-me-anyway.yaml:4:17: ' + trailing,
+            './s/s/ign-trail/s/s/file2.lint-me-anyway.yaml:5:5: ' + hyphen,
+        )))
```

Comment:
There are multiple things to enhance in these tests, for instance there is no control test (without `ignore` nor `ignore-from-file`) on the same data, and these similar tests would benefit from being gathered in the same class (to avoid reconstructing it each time).